servers/zts/src/main/java/com/yahoo/athenz/zts/notification/CertFailedRefreshNotificationTask.java

```diff
@@ -132,7 +132,7 @@ public CertFailedRefreshNotificationTask(InstanceCertManager instanceCertManager
 
     private List<X509CertRecord> getRecordsWithValidHosts(List<X509CertRecord> unrefreshedCerts) {
         return unrefreshedCerts.stream()
-                    .filter(record -> hostnameResolver.isValidHostname(record.getHostName()))
+                    .filter(record -> hostnameResolver == null || hostnameResolver.isValidHostname(record.getHostName()))
```

Comment:
Note that I've changed this check to only verify hostname only If hostnameResolver is used.